torch/distributed/launch.py

```diff
@@ -246,14 +269,31 @@ def main():
 
         cmd.extend(args.training_script_args)
 
-        process = subprocess.Popen(cmd, env=current_env)
+        subprocess_stdout = None
+        if args.logdir:
+            directory_path = os.path.join(os.getcwd(), args.logdir)
+            node_rank = args.node_rank
+            stdout_file_name = node_local_rank_stdout_filename.format(node_rank, local_rank)
+            file_path = os.path.join(directory_path, stdout_file_name)
+            file_handle = open(file_path, "w")
+            subprocess_stdout = file_handle
+            subprocess_file_handles.append(file_handle)
+            print(f"Note: Stdout for node {node_rank} rank {local_rank} will be written to {file_path}")
+
+
+        process = subprocess.Popen(cmd, env=current_env, stdout=subprocess_stdout)
```

Comment:
Shouldn't we also write stderr to a file?

Reply:
Updated the diff to do this.